cli/azd/pkg/tools/bicep/bicep_test.go

```diff
@@ -137,3 +139,75 @@ func TestNewBicepCliWillUpgrade(t *testing.T) {
 
 	require.Equal(t, []byte(NEW_FILE_CONTENTS), contents)
 }
+
+func Test_preferMuslBicep(t *testing.T) {
```

Comment:
Since I thought it was interesting, and I would share it with the team.  I actually generated most of this test with the help of Copilot Chat.  My prompt was:

> Write me a table drive test for the preferMuslBicep function.  There should be three cases, once where the stat call returns true for musl and false for glibc, one where it returns false for musl and true for glibc and one where it returns true for both.  The test should ensure that preferMuslBicep returns the correct value in each case.

I ended up having to make some changes to the generated test, but overall, I was impressed with the sketch that it gave me.

Reply:
I know that @weikanglim loves the `table drive` tests;  in my case, I don't :)

Is there a way to run each test from the table individually?  

This is of course, a nit, and thank you for sharing the prompt <3.

Reply:
> Is there a way to run each test from the table individually?

Yes, there is.  There are two options:

1. If you are running `go test` by hand, you can use the `-run` flag, to provide a filter for test names.  Since subtests work by appending / and then test name, you could do something like:

```
matell@Matts-Work-MacBook bicep% pwd 
/Users/matell/dd/azure/azure-dev/cli/azd/pkg/tools/bicep
matell@Matts-Work-MacBook bicep%  go test -run Test_preferMuslBicep/musl_preferred .
ok  	github.com/azure/azure-dev/cli/azd/pkg/tools/bicep	0.119s
```

The other option, which is a little more friendly (and makes things easier to debug) is to use the Testing pane in VS Code.  You can go to the package test and then run everything.  At that point, you should see the subtests show up (it has to run once for VS Code to understand what subtests exist) but you can see individual results of any subtest and debug that exact invocation if you want (in this screenshot, I hacked the test so one of the subtests failed):

![image](https://github.com/Azure/azure-dev/assets/9602953/a5a1841c-2983-4fed-beea-4893dfc9c133)

Once you run the tests once in VS, the `Test_preferMuslBicep` will turn into something that can be expanded.  You can right click on any subtest and do a run or debug.

I think this feature in VS Code is new within the last few months, I don't remember it previously being this easy to run individual subtests, but it is pretty nice now!

Hopefully the ergonomics are nice enough that you like table driven tests a little more now 😄.